src/compiler/checker.ts

```diff
@@ -32569,6 +32569,10 @@ namespace ts {
             if (target.kind === SyntaxKind.BinaryExpression && (target as BinaryExpression).operatorToken.kind === SyntaxKind.EqualsToken) {
                 checkBinaryExpression(target as BinaryExpression, checkMode);
                 target = (target as BinaryExpression).left;
+                // A default value is specified, so remove undefined from the final type.
+                if (strictNullChecks) {
+                    sourceType = getTypeWithFacts(sourceType, TypeFacts.NEUndefined);

```

Comment:
I think this may be broken if the default value is `undefined`—can you add a test?

Reply:
@andrewbranch Thank you for review!
[I added test cases with `undefined` default value](https://github.com/microsoft/TypeScript/pull/41042/files#diff-a73e59ec2a2848f7dc24482ff7f07936d54425791ebd895a1b909b46b8de4c06R18-R30).
Fortunately the current implementation seems OK.